platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+                        }
+
+                        @Override
+                        public String decrypt(String encryptedPassword) {
+                            return encryptionService.decrypt(encryptedPassword);
+                        }
+                    };
+                    encryptionCrypto = CryptoFactory
+                            .getInstance(sigProperties, IdpEndpoint.class.getClassLoader(),
+                                    passwordEncryptor);
+                } catch (WSSecurityException e) {
+                    LOGGER.debug("Error in loading the signature Crypto object: ", e);
+                }
+            } else {
+                LOGGER.debug("Cannot load signature properties using: " + encryptionPropertiesPath);
+            }
+        }
+        return encryptionCrypto;
+    }
+
+    protected String getSignaturePassword() {
+        if (signaturePropertiesPath != null) {
+            Properties sigProperties = PropertiesLoader.loadProperties(signaturePropertiesPath);
+            if (sigProperties != null) {
+                String password = sigProperties
+                        .getProperty(Merlin.PREFIX + Merlin.KEYSTORE_PRIVATE_PASSWORD);
+                if (password == null) {
+                    password = sigProperties
+                            .getProperty(Merlin.OLD_PREFIX + Merlin.KEYSTORE_PRIVATE_PASSWORD);
+                }
+                if (password != null) {
+                    password = password.trim();
+                    password = decryptPassword(password, encryptionService);
+                }
+                return password;
+            }
+        }
+        return null;
+    }
+
+    protected String getEncryptionPassword() {
+        if (encryptionPropertiesPath != null) {
+            Properties encProperties = PropertiesLoader.loadProperties(encryptionPropertiesPath);
+            if (encProperties != null) {
+                String password = encProperties
+                        .getProperty(Merlin.PREFIX + Merlin.KEYSTORE_PRIVATE_PASSWORD);
+                if (password == null) {
+                    password = encProperties
+                            .getProperty(Merlin.OLD_PREFIX + Merlin.KEYSTORE_PRIVATE_PASSWORD);
+                }
+                if (password != null) {
+                    password = password.trim();
+                    password = decryptPassword(password, encryptionService);
+                }
+                return password;
+            }
+        }
+        return null;
+    }
+
+    protected String decryptPassword(String password, EncryptionService passwordEncryptor) {
+        if (password.startsWith(Merlin.ENCRYPTED_PASSWORD_PREFIX) && password
+                .endsWith(Merlin.ENCRYPTED_PASSWORD_SUFFIX)) {
+            if (passwordEncryptor == null) {
+                LOGGER.debug("No PasswordEncryptor is configured!");
+                return password;
+            }
+            String substring = password
```

Comment:
Looks right, but a bit fragile with the risk of a fencepost error. Try `StringUtils.removeStart()` and `StringUtils.removeEnd()`.